routes/user.js

```diff
@@ -0,0 +1,176 @@
+var app = express();
+const bcrypt = require("bcrypt");
+var { auth } = require("../config/auth");
+var { user } = require("../models/User");
+var { AccessToken } = require("../models/AccessToken");
+
+app.post("/register", async function (req, res) {
+  const salt = await bcrypt.genSalt();
+  const userPassword = await bcrypt.hash(req.body.password, salt);
+
+  if (req.body.password !== req.body.confirmpassword) {
+    res.status(500).json({
+      error: 1,
+      message: "password not matched",
+      data: null,
+    });
+  } else {
+    const user_post = new user({
+      username: req.body.username,
+      firstname: req.body.firstname,
+      lastname: req.body.lastname,
+      password: userPassword,
+      email: req.body.email,
+    });
+    user_post.save(function (err, row) {
+      if (err) {
+        res.status(500).json({
+          error: 1,
+          message: err.message,
+          data: null,
+        });
+      } 
+      else {
+        res.status(200).json({
+          error: 0,
+          message: "saved successfully",
+          data: null,
+        });
+      }
+    });
+  }
+});
+
+app.post("/login", async function (req, res) {
+  user.findOne({ username: req.body.username }, async function (err, results) {
+    var pass = results.password.toString();
+    var userId = results._id;
+    var input_password = pass.toString();
+    var user_password = req.body.password;
+    var tokenId = userId.toString();
+    if (await bcrypt.compare(user_password, input_password)) {
+      const pwd = passwordHash.generate(req.body.password);
+      AccessToken.findOne({ user_id: userId }, function (err, userDetails) {
+        if (userDetails && userDetails._id) {
+          res.status(500).json({
+            error: 1,
+            message: "already login",
+            data: null,
+          });
+        } else {
+          var token_save = new AccessToken({
+            user_id: userId,
+            access_token: pwd,
+          });
+          token_save.save(function (err) {
+            if (err) {
+              res.status(200).json({
+                error: 1,
+                message: "token not saved",
+                data: null,
+              });
+            }
+            res.status(200).json({
+              error: 0,
+              message: "token saved",
+              data: tokenId,
+            });
+          });
+        }
+      });
+    } else {
+      res.status(500).json({
+        error: 1,
+        message: "no user found",
+        data: null,
+      });
+    }
+  });
+});
+
+app.get("/get", auth, async function (req, res) {
+  res.status(200).json({
+    error: 0,
+    message: "user list",
+    data: {
+      id: req.user._id,
+      username: req.user.username,
+      email: req.user.email,
+    },
+  });
+});
+
+app.put("/delete/", auth, async function (req, res) {
+  var user_id = req.user.user_id;
+  await user.deleteOne({ _id: user_id }, function (err, results) {
```

Comment:
- don't use await and callback at same time.
- delete user address after user delete